src/Compilers/CSharp/Portable/CodeGen/EmitAddress.cs

```diff
@@ -535,17 +564,17 @@ private LocalDefinition EmitReceiverRef(BoundExpression receiver, bool isAccessC
             }
 
             Debug.Assert(receiverType.IsVerifierValue());
-            return EmitAddress(receiver, AddressKind.Writeable);
+            return EmitAddress(receiver, addressKind);
         }
 
         /// <summary>
         /// May introduce a temp which it will return. (otherwise returns null)
         /// </summary>
-        private LocalDefinition EmitInstanceFieldAddress(BoundFieldAccess fieldAccess)
+        private LocalDefinition EmitInstanceFieldAddress(BoundFieldAccess fieldAccess, bool isReadonly)
```

Comment:
This is subtle. I will add a comment why AddressKind should not be propagated.

---
In reply to: [108271480](https://github.com/dotnet/roslyn/pull/18175#discussion_r108271480) [](ancestors = 108271480,108265271)